wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt

```diff
@@ -311,17 +333,51 @@ class EditProfileActivity : BaseMenuActivity() {
                 }
                 REQUEST_CODE_CROP_IMAGE -> {
                     if (resultCode == Activity.RESULT_OK) {
-                        setAvatarFromFile(editProfileViewModel.profilePictureFile!!)
+                        if (externalUrlSharedViewModel.externalUrl != null) {
+                            saveUrl(CropImageActivity.extractZoomedRect(data!!))
+                        } else {
+                            setAvatarFromFile(editProfileViewModel.profilePictureFile!!)
+                        }
                     }
                 }
             }
         }
     }
 
+    private fun saveUrl(zoomedRect: RectF) {
+        if (externalUrlSharedViewModel.externalUrl != null) {
+            val zoomedRectStr = "${zoomedRect.left},${zoomedRect.top},${zoomedRect.right},${zoomedRect.bottom}"
+            externalUrlSharedViewModel.externalUrl = setUriParameter(externalUrlSharedViewModel.externalUrl!!, "dashpay-profile-pic-zoom", zoomedRectStr)
+
+            val file = editProfileViewModel.tmpPictureFile
+            val imgUri = getFileUri(file)
+            Glide.with(dashpayUserAvatar).load(imgUri)
+                    .signature(ObjectKey(file.lastModified()))
+                    .placeholder(defaultAvatar)
+                    .transform(ProfilePictureTransformation.create(zoomedRect))
+                    .into(dashpayUserAvatar)
+        }
+    }
+
+    private fun setUriParameter(uri: Uri, key: String, newValue: String): Uri {
+        val newUriBuilder = uri.buildUpon()
+        if (uri.getQueryParameter(key) == null) {
+            newUriBuilder.appendQueryParameter(key, newValue)
+        } else {
+            newUriBuilder.clearQuery()
+            for (param in uri.queryParameterNames) {
+                newUriBuilder.appendQueryParameter(param,
+                        if (param == key) newValue else uri.getQueryParameter(param))
+            }
+        }
+        return newUriBuilder.build()
+    }
+
     private fun cropProfilePicture() {
         val tmpPictureUri = editProfileViewModel.tmpPictureFile.toUri()
         val profilePictureUri = editProfileViewModel.profilePictureFile!!.toUri()
-        val intent = CropImageActivity.createIntent(this, tmpPictureUri, profilePictureUri)
+        val initZoomedRect = ProfilePictureTransformation.extractZoomedRect(externalUrlSharedViewModel.externalUrl)
```

Comment:
`externalUrlSharedViewModel.externalUrl` is null when selecting a picture from Camera or Device.

Reply:
Fixed